src/beaker_harness.c

```diff
@@ -0,0 +1,71 @@
+/*
+    This file is part of Restraint.
+
+    Restraint is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    Restraint is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with Restraint.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include <stdlib.h>
+#include <string.h>
+#include <glib.h>
+#include <libsoup/soup.h>
+
+#include "beaker_harness.h"
+
+#define BKR_ENV_FILE "/etc/profile.d/beaker-harness-env.sh"
+
+#define STREMPTY(str) (str == NULL || strlen (str) == 0)
+
+static int rstrnt_bkr_env_file_exists (const char *env_file);
+
+int
```

Comment:
Shouldn't these function return a gboolean?

Reply:
I'm avoiding GLib for the shared code.